xgcm/test/test_metrics.py

```diff
@@ -96,6 +97,42 @@ def test_assign_metric(key, metric_vars):
     _ = Grid(ds, coords=coords, metrics={key: metric_vars})
 
 
+@pytest.mark.parametrize(
+    "axes, expected_axes",
+    [
+        (
+            ("X", "Y"),
+            (
+                (frozenset({"X", "Y"}),),
+                (frozenset({"X"}), frozenset({"Y"})),
+                (frozenset({"Y"}), frozenset({"X"})),
+            ),
+        ),
+        (
+            ("X", "Y", "Z"),
+            (
+                (frozenset({"Y", "X", "Z"}),),
+                (frozenset({"Z"}), frozenset({"Y"}), frozenset({"X"})),
+                (frozenset({"Z"}), frozenset({"X"}), frozenset({"Y"})),
+                (frozenset({"X"}), frozenset({"Y"}), frozenset({"Z"})),
+                (frozenset({"X"}), frozenset({"Z"}), frozenset({"Y"})),
+                (frozenset({"Y"}), frozenset({"X"}), frozenset({"Z"})),
+                (frozenset({"Y"}), frozenset({"Z"}), frozenset({"X"})),
+                (frozenset({"Y", "Z"}), frozenset({"X"})),
+                (frozenset({"Y", "X"}), frozenset({"Z"})),
+                (frozenset({"X", "Z"}), frozenset({"Y"})),
+                (frozenset({"X", "Y"}), frozenset({"Z"})),
+            ),
+        ),
+    ],
+)
+def test_iterate_axis_combinations(axes, expected_axes):
+
+    iterate_axes = set(iterate_axis_combinations(axes))
+    set_diff = iterate_axes.difference(expected_axes)
+    assert set_diff == set()
```

Comment:
We should be able to directly check equality?

```suggestion
    assert iterate_axes == expected_axes
```

Reply:
When I use `assert iterate_axes == expected_axes`, I get the ff error:
```python
>       assert iterate_axes == expected_axes
E       AssertionError: assert {(frozenset({...{'X', 'Y'}),)} == ((frozenset({...enset({'X'})))
E         Use -v to get the full diff
```
But when I use set difference the test passes. Does it pass for you if you directly check equality?

Reply:
Ah two things.

1. `expected_axes` is a tuple so `assert iterate_axes == set(expected_axes)` works for the first case.
2. The set difference is not symmetric! `iterate_axes.difference(expected_axes)` will be empty if there are no elements in `iterate_axes` that are not in `expected_axes`. But `expected_axes` could have extra elements and that difference would still be an empty set. You could use `symmetric_difference`. But it should fail just as direct equality fails for the second set. 

So the second set of parameters needs to be fixed.

Reply:
I see, thanks Deepak!

Reply:
I get AssertionErrors because when I run iterate_axis_combinations it will produce only one of this type of combination, and `assert iterate_axes == set(expected_axes)` doesn't count these as equal:
```python
(frozenset({'Y'}), frozenset({'Z'}), frozenset({'X'}))
(frozenset({'Y'}), frozenset({'X'}), frozenset({'Z'}))
```
and you can't predict which one iterate_axis_combination will produce so setting up `expected_axes` feels like whack-a-mole as I try to get the left and right set to match. 😅 I'll try to see what can be a more efficient way to test this.

Reply:
I think that means you want to convert things to a set one level down. So that a single element is `set(frozenset({'Y'}), frozenset({'Z'}), frozenset({'X'}))`

Reply:
If I change an element into a set: `set((frozenset({"Z"}), frozenset({"Y"}), frozenset({"X"})))`
There's a TypeError: 
```python 
E       TypeError: unhashable type: 'set'
```
if I do: `set(frozenset({"Z"}), frozenset({"Y"}), frozenset({"X"}))`
I also get a TypeError:
```python
E   TypeError: set expected at most 1 argument, got 3
```

Reply:
I think the main issue is that `iterate_axis_combinations` will always produce a set of tuples. So an element of the result (of type tuple) can either be:
```python
(frozenset({"Z"}), frozenset({"X"}), frozenset({"Y"}))
(frozenset({"Z"}), frozenset({"Y"}), frozenset({"X"}))
```
And since tuples are compared position by position, `expected_axes` can only contain one of the above so that
```python
assert iterate_axes == set(expected_axes)
```
will result in `True`. Unfortunately, there's no real way to predict what `iterate_axis_combinations` will produce, especially if you factor in the other combinations such as `(frozenset({"Y"}), frozenset({"X"}), frozenset({"Z"})),` and `(frozenset({"X"}), frozenset({"Y"}), frozenset({"Z"})),`. We can discuss this further, but I think the solution will be to keep the test limited to `axes = ("X","Y")`.

Reply:
Alright I fixed it a hacky way. Now we can move on to bigger and better things =)